### PR TITLE
Pxb 2.4 5.7.40 merge 2990

### PR DESCRIPTION
**Pull request by VarunNagaraju**

### CODE REVIEW

File: include/mysql/psi/mysql_thread.h [hunk 1]
```diff
@@ -1320,15 +1320,15 @@ static inline void inline_mysql_thread_set_psi_THD(THD *thd)
 }
 #endif /* __cplusplus */
 
+#endif
```
Comment (rahulmalik87) [inline include/mysql/psi/mysql_thread.h]: can you comment that is closer of which if

**Review by rahulmalik87 — CHANGES_REQUESTED** [review]
Please test in PXB jenkins as well.

File: storage/innobase/fil/fil0fil.cc [hunk 2]
```diff
@@ -1561,17 +1561,10 @@ fil_space_is_being_truncated(
 	bool flag = true;
 	fil_space_t* space;
 	mutex_enter(&fil_system->mutex);
-<<<<<<< HEAD
-	fil_space_t* space = fil_space_get_by_id(id);
-	mark_for_truncate = space ? space->is_being_truncated : false;
```
Comment (altmannmarcelo) [inline storage/innobase/fil/fil0fil.cc:1547]: we already had this fix for 5 years :D

File: include/mysql/psi/mysql_thread.h (hunk 1 above)
Comment (altmannmarcelo) [reply include/mysql/psi/mysql_thread.h]: Why we are moving up the enclose ifdef HAVE_PSI_THREAD_INTERFACE ?

PXB should have it as false and do not compile this part of code.

Also, please include in the commit message which compilation error you were getting that lead to this fix.

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]

**Comment by rahulmalik87** [comment]
Please paste Jenkin's result  link

**Comment by VarunNagaraju** [comment]
> Please paste Jenkin's result link

https://pxb.cd.percona.com/view/all/job/percona-xtrabackup-2.4-param/135/

**Comment by VarunNagaraju** [comment]
Jenkins results: https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param/139

**Review by rahulmalik87 — APPROVED** [review]
LGTM

**Review by altmannmarcelo — APPROVED** [review]
LGTM

**Pull request merged**